HubSpot.NET/Core/Interfaces/IHubSpotCompanyApi.cs

```diff
@@ -3,13 +3,13 @@
 
 namespace HubSpot.NET.Core.Interfaces
 {
-    public interface IHubSpotCompanyApi
+    public interface IHubSpotCompanyApi : IHubSpotCompanyApi<CompanyHubSpotModel>
+    { }
+
+    public interface IHubSpotCompanyApi<T> : ICRUDable<T>
```

Comment:
Ah, so is this how we enable the use of custom models for contacts, e.g. to use your own properties etc. I'll pull this down and have a play but could you explain how this works in the meantime?

This is obviously a big breaking change to the API so we'll need to have a think about how we version this. I'm tempted to pull this into a 'v1' branch and iterate/churn on it there before releasing it as a v1 major new release? What do you think?

Reply:
On you first question, are you asking how the inherited interfaces are working?

Yes, this is definitely a breaking change. But I think our clients will be the better for it. I definitely think we are in the `v1` realm now for sure. We are currently dogfooding this version ourselves and it is working pretty solidly, but I do agree that having more use cases thrown at the SDK will make it more robust. You might have noticed that we made plenty of version updates to the `.nuspec` file, but that was due to our private build agent failing if we didn't. We might even make a `v1-preview` or `v1-beta` version to release and let others help us break it and fix the bugs we haven't caught. This would also to give us some time to add Unit Tests.

Are there any more features you are hoping to have included in a `v1`?

Reply:
Right, so the idea here is to set the conditions for the interface/contract while also reducing the number of files repeating the same pattern just to gain strongly typed definitions. So the generic T is shared by each of the sub-interfaces such that you only have to pass the definition for T into the top level interface to gain the benefit of strongly typed contracts for each of the lower level interfaces. Layering interfaces together like this is also how the Entity Framework team provides strongly typed databases and DBSets.

Reply:
I have a deep appreciation for layered interfaces. Haha